### PR TITLE
Don't include appservice users when calculating push rules

### PR DESCRIPTION
**Pull request by erikjohnston**
This can cause a lot of extra load on servers with lots of appservice users. Introduced in https://github.com/matrix-org/synapse/pull/13078

### CODE REVIEW

**Review by MatMaul — APPROVED** [review]
LGTM.

**Comment by clokep** [comment]
Is this something we should have a unit test for?

**Comment by erikjohnston** [comment]
Testing is cheating!

(yes we should)

File: changelog.d/13332.bugfix [hunk 1]
```diff
@@ -0,0 +1 @@
+Fix bug where we incorrectly calculated push for appservice users, leading to performance issues on servers with large numbers of appservics. Introduced in v1.63.0.
```
Comment (DMRobertson) [inline changelog.d/13332.bugfix:1]:
```suggestion
Fix a bug introduced in Synapse 1.63.0 where push actions were incorrectly calculated for appservice users. This caused performance issues on servers with large numbers of appservices.
```

**Review by DMRobertson — APPROVED** [review]
SGTM otherwise. Shame we (I?) missed this in review.

Do we want to do a 1.63.1 release with this fix?

**Pull request merged**